boards/feather_m0/examples/ssd1306_graphicsmode_128x32_i2c.rs

```diff
@@ -0,0 +1,156 @@
+//!
+//! You can either use the USB port on the Feather + BOSSAC or a SWD
+//! debugger/programmer to load this example onto the Feather M0 board.
+//!
+//! Adafruit Feather M0 Boards:
+//! - Feather M0 Express: https://www.adafruit.com/product/3403
+//! - Feather M0 Adalogger: https://www.adafruit.com/product/2796
+//! - Feather M0 Basic: https://www.adafruit.com/product/2772
+//!
+//! Adafruit OLED Featherwing:
+//! - https://www.adafruit.com/product/2900
+//!
+//! Other generic SSD1306 modules will work with this demo as well.
+//!
+//! Wiring connections for the Adafruit OLED Featherwing:
+//!
+//! ```
+//! OLED Featherwing -> Feather M0
+//!     GND -> GND
+//!     3v  -> 3v3
+//! GPIOSDA -> PA22 (SDA)
+//! GPIOSCL -> PA23 (SCL)
+//! ```
+//!
+//! Build this example with: `cargo build --example ssd1306_graphicsmode_128x32_i2c`
+//!
+
+extern crate feather_m0 as hal;
+extern crate embedded_graphics;
+extern crate ssd1306;
+
+// how to panic...
+#[cfg(not(feature = "use_semihosting"))]
+extern crate panic_halt;
+#[cfg(feature = "use_semihosting")]
+extern crate panic_semihosting;
+
+use hal::clock::GenericClockController;
+use hal::delay::Delay;
+use hal::prelude::*;
+use hal::time::KiloHertz;
+use hal::entry;
+use hal::pac::{CorePeripherals, Peripherals};
+
+use embedded_graphics::prelude::*;
+use embedded_graphics::primitives::{Circle, Line, Rectangle};
+use embedded_graphics::pixelcolor::BinaryColor;
+use ssd1306::prelude::*;
+use ssd1306::Builder;
+
+#[entry]
+fn main() -> ! {
+    let mut peripherals = Peripherals::take().unwrap();
+    let core = CorePeripherals::take().unwrap();
+    let mut clocks = GenericClockController::with_internal_32kosc(
+        peripherals.GCLK,
+        &mut peripherals.PM,
+        &mut peripherals.SYSCTRL,
+        &mut peripherals.NVMCTRL,
+    );
+    let mut pins = hal::Pins::new(peripherals.PORT);
+    let mut red_led = pins.d13.into_open_drain_output(&mut pins.port);
+    let mut delay = Delay::new(core.SYST, &mut clocks);
+
+    let i2c = hal::i2c_master(
+        &mut clocks,
+        KiloHertz(400),
+        peripherals.SERCOM3,
+        &mut peripherals.PM,
+        pins.sda,
+        pins.scl,
+        &mut pins.port,
+    );
+
+    // NOTE the `DisplaySize` enum comes from the ssd1306 package,
+    // and currently only supports certain display sizes; see
+    // https://jamwaffles.github.io/ssd1306/master/ssd1306/prelude/enum.DisplaySize.html
+    let mut disp: GraphicsMode<_> = Builder::new()
+        .size(DisplaySize::Display128x32)
+        .connect_i2c(i2c)
+        .into();
+
+    disp.init().unwrap();
+    disp.flush().unwrap();
+
+    let yoffset = 8;
+    let x_max = 127;
+    let y_max = 31;
+
+    // screen outline
+    disp.draw(
+        Rectangle::new(Point::new(0, 0), Point::new(x_max, y_max))
+            .stroke(Some(BinaryColor::On))
+            .into_iter(),
+    );
+
+    // triangle
```

Comment:
Embedded Graphics has a new [`Triangle` primitive](https://docs.rs/embedded-graphics/0.6.0-alpha.2/embedded_graphics/primitives/triangle/struct.Triangle.html) which could replace the 3 lines drawn below FWIW.

Reply:
I'll take a poke at it and see what happens.